src/wasm/asm-wasm-builder.cc

```diff
@@ -174,11 +196,19 @@ class AsmWasmBuilderImpl : public AstVisitor {
 
   void VisitFunctionLiteral(FunctionLiteral* expr) {
     Scope* scope = expr->scope();
-    // TODO: confirm where to get type
     if (in_function_) {
-      current_function_builder_->ReturnType(kAstI32);
-      for (int i = 0; i < expr->parameter_count(); i++) {
-        LookupOrInsertLocal(scope->parameter(i), true);
+      if (expr->bounds().lower->IsFunction()) {
+        Type::FunctionType* func_type = expr->bounds().lower->AsFunction();
+        LocalType return_type = TypeFrom(func_type->Result());
+        DCHECK(return_type != kAstStmt);
```

Comment:
done. No tests added yet as I haven't added globals or instructions for working off heap yet.